extensions/mssql/src/dashboard/bookWidget.ts

```diff
@@ -71,58 +67,21 @@ export function registerBooksWidget(bookContributionProvider: BookContributionPr
 		});
 		await view.initializeModel(bookslocationContainer);
 	});
+
+	registerCommands(bookContributionProvider);
 }
 
-async function promptForFolder(bookContribution: BookContribution): Promise<void> {
-	try {
-		const allFilesFilter = localize('allFiles', "All Files");
-		let filter = {};
-		filter[allFilesFilter] = '*';
-		let uris = await vscode.window.showOpenDialog({
-			filters: filter,
-			canSelectFiles: false,
-			canSelectMany: false,
-			canSelectFolders: true,
-			openLabel: localize('labelPickFolder', "Pick Folder")
-		});
-		if (uris && uris.length > 0) {
-			let pickedFolder = uris[0];
-			let destinationUri: vscode.Uri = vscode.Uri.file(path.join(pickedFolder.fsPath, bookContribution.name));
-			await saveBooksToFolder(destinationUri, bookContribution);
-			await promptToReloadWindow(destinationUri);
-		}
-		return;
-	} catch (error) {
-		vscode.window.showErrorMessage(localize('FailedDuringSaveAndPrompt', 'Failed : {0}', Utils.getErrorMessage(error)));
-	}
+function openBookViewlet(folderUri: vscode.Uri): void {
+	vscode.commands.executeCommand('bookTreeView.openBook', folderUri.fsPath, true);
 }
 
-async function saveBooksToFolder(folderUri: vscode.Uri, bookContribution: BookContribution): Promise<void> {
-	// Get book contributions
-	if (bookContribution && folderUri) {
-		//remove folder if exists
-		await fs.removeSync(folderUri.fsPath);
-		//make directory for each contribution book.
-		await fs.mkdirSync(folderUri.fsPath);
-		await fs.copy(bookContribution.path, folderUri.fsPath);
+function registerCommands(bookContributionProvider: any): void {
+	let sql19Preview: boolean = false;
+	if (bookContributionProvider.contributions.filter(books => books.name === 'SQL Server 2019 Guide').length > 0) {
```

Comment:
Isn't this value going to be localized? That won't work with this comparison then. 

Where is this book contributed anyways? It seems like whatever is doing that should also be registering this command - not relying on the widget to do that work.

Reply:
Don't know what I was thinking when I moved this here. Corrected it and moved it to the book contribution.